mardown_converter.py

```
@@ -1,40 +1,82 @@
+  // Wrap unordered lists — only first match
   html = html.replace(/(<li>.*<\/li>)/, '<ul>$1</ul>');
 
-  // paragraph wrapping — overly greedy, swallows multiple lines
-  html = html.replace(/^(?!<h\d|<ul|<li|<blockquote|<code|<a)(.+)$/gms, '<p>$1</p>');
+  // Wrap ordered lists — completely broken, mismatched tags
+  html = html.replace(/<li>(.*?)<\/li>/, '<ol>$1</ol>');
+
+  // Wrap paragraphs — greedy regex across multiple lines
+  html = html.replace(/^(?!<h\d|<ul|<ol|<li|<blockquote|<code|<a|<img|<hr)(.+)$/gms, '<p>$1</p>');
+
+  // Attempt to handle fenced code blocks (```), but wrong regex
```

Ruff (0.12.2)

SyntaxError: Expected a statement

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Expected ',', found name

---

SyntaxError: Expected ',', found name

---

SyntaxError: Expected an expression or a ')'

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Expected an expression

---

SyntaxError: Expected an identifier

---

SyntaxError: Expected an expression

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Got unexpected token `

---

SyntaxError: Expected ',', found '/'

---

Multiple statements on one line (semicolon)

(E702)